pkg/aspect/logger/logger.go

```diff
@@ -30,10 +32,23 @@ type (
 		Log([]Entry) error
 	}
 
-	// Entry is a set of key-value pairs of attributes that together
-	// constitute the data for log entry. The key is the attribute name,
-	// and the value is a typed value for the named attribute.
-	Entry map[string]interface{}
+	// Entry is the set of data that together constitutes a log entry.
+	Entry struct {
+		// LogName is the name of the log in which to record this entry.
+		LogName string
+		// Labels are a set of metadata associated with this entry.
+		// For instance, Labels can be used to describe the monitored
+		// resource that corresponds to the log entry.
+		Labels map[string]interface{}
+		// Payload is the actual data for which the entry is being
+		// generated.
+		Payload string
+		// Timestamp is the time value for the log entry
+		Timestamp time.Time
+		// Severity indicates the log level for the log entry.
+		// Example levels: "INFO", "WARNING", "500"
+		Severity string
```

Comment:
Given our discussion, what we want at the moment is to lock down the API to the aspect. So all incoming parameters are well-known and fully specified, no escape hatch. The job of the adapter is to map from this generic API to what its backend expects. The mapping might be lossy and that's OK.